File: app/src/main/java/com/example/final_pocproject/Retrofit/ApiInterface.java
package com.example.final_pocproject.Retrofit;

import com.example.final_pocproject.Model.CommentList;
import com.example.final_pocproject.Model.Post;
import com.example.final_pocproject.Model.User;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface ApiInterface {

    @GET("users")
    Call<List<User>> getUsers();

    @GET("users/{id}/posts")
    Call <List<Post>> getPosts(@Path("id") int postId);

    @GET("posts/{id}/comments")
    Call <List<CommentList>> getComments(@Path("id") int postId);

}

File: app/src/main/java/com/example/final_poc_project/retrofit/ApiInterface.kt
package com.example.final_poc_project.retrofit

import com.example.final_poc_project.model.Comment
import com.example.final_poc_project.model.Post
import com.example.final_poc_project.model.User
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface ApiInterface {

    @GET("users")
    fun getUsers(): Call<List<User>>

    @GET("users/{id}/posts")
    fun getPosts(@Path("id") userId: Int): Call<List<Post>>

    @GET("posts/{id}/comments")
    fun getComments(@Path("id") postId: Int): Call<List<Comment>>
}